Automattic/wp-calypso
# Nav redesign: Refactor to simplify routing logic

## Proposed Changes

This PR refactors the logic introduced in https://github.com/Automattic/wp-calypso/pull/89949:

1.  Render the whole route instead of duplicating every routing logic in `createFeaturePreview()`:

    <img width="393" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/a4a235ce-d156-4cf3-ba77-e4b3aa9e825e">

    <img width="1014" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/5e3d0255-a7d8-4f72-9cd2-544f117cd8c3">

    with this, when (for example) GH Deployments add another route, we don't need to worry about it.

3. To better support the above, extract the route <=> URL syncing logic into their own hooks for better readability:

    <img width="655" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/85cfe53f-a281-4e5d-99ab-e897cc179907">

4. Duplicated the controller/routes from the old sites dashboard v1 so that we're decoupled (as suggested by @cleacos here: https://github.com/Automattic/wp-calypso/pull/89979)

    <img width="514" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/2c44964b-5f37-4051-a808-00174e284b86">

## Caveats

We still have this "hack" of putting the route as feature IDs (see: https://github.com/Automattic/wp-calypso/pull/89949#discussion_r1581579446). We can resolve this later if needed. cc: @cleacos 

## Testing Instructions



1. Go to `<Calypso Live URL>/sites`
    - Verify that the old sites dashboard v1 works normally.
1. Go to `<Calypso Live URL>/sites?layout/dotcom-nav-redesign-v2`
    - Try to break it / find any regression from Horizon's.

## Pre-merge Checklist
